syntax = "proto3";

option csharp_namespace = "FSR.GRPC.Lib.V3.Services.AssetAdministrationShellRepository";

import "Protos/AasModels.V3.proto";
import "Protos/AasApiModels.V3.proto";

package FSR.GRPC.Lib.V3;

/* AssetAdministrationShellRepositoryService */

message GetAllAssetAdministrationShellsRpcRequest {
    OutputModifier outputModifier = 1;
}
message GetAllAssetAdministrationShellsRpcResponse {
    int32 statusCode = 1;
    PagedResultPagingMetadata pagingMetaData = 2; 
    repeated AssetAdministrationShellDTO payload = 3;
    repeated ReferenceDTO reference = 4;
}

message GetAssetAdministrationShellByIdRpcRequest {
    string id = 1;
    OutputModifier outputModifier = 2;
}
message GetAssetAdministrationShellByIdRpcResponse {
    int32 statusCode = 1;
    AssetAdministrationShellDTO payload = 2;
    ReferenceDTO reference = 3;
}

message GetAllAssetAdministrationShellsByAssetIdRpcRequest {
    string key = 1;
    string keyIdentifier = 2;
    OutputModifier outputModifier = 3;
}
message GetAllAssetAdministrationShellsByAssetIdRpcResponse {
    int32 statusCode = 1;
    PagedResultPagingMetadata pagingMetaData = 2; 
    repeated AssetAdministrationShellDTO payload = 3;
    repeated ReferenceDTO reference = 4;
}

message GetAllAssetAdministrationShellsByIdShortRpcRequest {
    string idShort = 1;
    OutputModifier outputModifier = 2;
}
message GetAllAssetAdministrationShellsByIdShortRpcResponse {
    int32 statusCode = 1;
    PagedResultPagingMetadata pagingMetaData = 2; 
    repeated AssetAdministrationShellDTO payload = 3;
    repeated ReferenceDTO reference = 4;
}

message PostAssetAdministrationShellRpcRequest {
    AssetAdministrationShellDTO aas = 1;
}
message PostAssetAdministrationShellRpcResponse {
    int32 statusCode = 1;
    AssetAdministrationShellDTO aas = 2;
}

message PutAssetAdministrationShellByIdRpcRequest {
    AssetAdministrationShellDTO aas = 1;
}
message PutAssetAdministrationShellByIdRpcResponse {
    int32 statusCode = 1;
    AssetAdministrationShellDTO aas = 2;
}

message DeleteAssetAdministrationShellByIdRpcRequest {
    string id = 1;
}
message DeleteAssetAdministrationShellByIdRpcResponse {
    int32 statusCode = 1;
}

service AssetAdministrationShellRepositoryService {
    rpc GetAllAssetAdministrationShells (GetAllAssetAdministrationShellsRpcRequest) returns (GetAllAssetAdministrationShellsRpcResponse);
    rpc GetAssetAdministrationShellById (GetAssetAdministrationShellByIdRpcRequest) returns (GetAssetAdministrationShellByIdRpcResponse);
    rpc GetAllAssetAdministrationShellsByAssetId (GetAllAssetAdministrationShellsByAssetIdRpcRequest) returns (GetAllAssetAdministrationShellsByAssetIdRpcResponse);
    rpc GetAllAssetAdministrationShellsByIdShort (GetAllAssetAdministrationShellsByIdShortRpcRequest) returns (GetAllAssetAdministrationShellsByIdShortRpcResponse);
    rpc PostAssetAdministrationShell (PostAssetAdministrationShellRpcRequest) returns (PostAssetAdministrationShellRpcResponse);
    rpc PutAssetAdministrationShellById (PutAssetAdministrationShellByIdRpcRequest) returns (PutAssetAdministrationShellByIdRpcResponse);
    rpc DeleteAssetAdministrationShellById (DeleteAssetAdministrationShellByIdRpcRequest) returns (DeleteAssetAdministrationShellByIdRpcResponse);
}